zerver/tests/test_markdown.py

```diff
@@ -2749,9 +2749,9 @@ def test_send_message_errors(self) -> None:
                 self.send_stream_message(self.example_user("othello"), "Denmark", message)
 
     def test_ultra_long_rendering(self) -> None:
-        """A rendered message with an ultra-long length (> 10 * MAX_MESSAGE_LENGTH)
+        """A rendered message with an ultra-long length (> 100 * MAX_MESSAGE_LENGTH)
         throws an exception"""
-        msg = "mock rendered message\n" * MAX_MESSAGE_LENGTH
+        msg = "mock rendered message\n" * 10 * MAX_MESSAGE_LENGTH
```

Comment:
I think we need to decrease MAX_MESSAGE_LENGTH for this test, because as written it takes 0.5s to run just this test.

The right idea might be to change `MAX_MESSAGE_LENGTH` to a setting defined in `default_settings.py` so that we can mock it with `with self.settings(MAX_MESSAGE_LENGTH=10` or whatever.

(That would also offer an undocumented interface for someone to adjust that value when self-hosting).

@akshatdalton would you be up for doing that preparatory PR to make `MAX_MESSAGE_LENGTH` a setting?

Reply:
@timabbott 
Opened PR: #18683.

Reply:
OK, merged that and rebased this on top of that; I intend to merge once CI passes.